src/amberscript/parser.cc

```diff
@@ -1307,6 +1330,150 @@ Result Parser::ParsePipelineDepth(Pipeline* pipeline) {
   return ValidateEndOfStatement("DEPTH command");
 }
 
+Result Parser::ParsePipelineStencil(Pipeline* pipeline) {
+  auto token = tokenizer_->NextToken();
+  if (!token->IsIdentifier())
+    return Result("STENCIL missing face");
+
+  bool setFront = false;
+  bool setBack = false;
+
+  if (token->AsString() == "front") {
+    setFront = true;
+  } else if (token->AsString() == "back") {
+    setBack = true;
+  } else if (token->AsString() == "front_and_back") {
+    setFront = true;
+    setBack = true;
+  } else {
+    return Result("STENCIL invalid face: " + token->AsString());
+  }
+
+  while (true) {
+    token = tokenizer_->NextToken();
+    if (token->IsEOL())
+      continue;
+    if (token->IsEOS())
+      return Result("STENCIL missing END command");
+    if (!token->IsIdentifier())
+      return Result("STENCIL options must be identifiers");
+    if (token->AsString() == "END")
+      break;
+
+    if (token->AsString() == "TEST") {
+      token = tokenizer_->NextToken();
+
+      if (!token->IsIdentifier())
+        return Result("STENCIL invalid value for TEST");
+
+      if (token->AsString() == "on")
+        pipeline->GetPipelineData()->SetEnableStencilTest(true);
+      else if (token->AsString() == "off")
+        pipeline->GetPipelineData()->SetEnableStencilTest(false);
+      else
+        return Result("STENCIL invalid value for TEST: " + token->AsString());
+    } else if (token->AsString() == "FAIL_OP") {
+      token = tokenizer_->NextToken();
+
+      if (!token->IsIdentifier())
+        return Result("STENCIL invalid value for FAIL_OP");
+
+      StencilOp stencil_op = StrToStencilOp(token->AsString());
+      if (stencil_op != StencilOp::kUnknown) {
```

Comment:
For this and the below let's handle the error case first, then we don't need the else.

```
if (stencil_op == StencilOp::kUnknown) {
  return Result ....;
}
```

Reply:
sure